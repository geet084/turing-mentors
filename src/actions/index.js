export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

export const updateForm = (form) => ({
  type: 'UPDATE_FORM',
  form
});

export const resetForm = () => ({
  type: 'RESET_FORM',
})

export const getMentorsSuccess = (mentors) => ({
  type: 'GET_MENTORS_SUCCESS',
  mentors
});

export const slackSuccess = (message) => ({
  type: 'SLACK_SUCCESS',
  message
});

export const postMentorSuccess = (mentor) => ({
  type: 'POST_MENTOR_SUCCESS',
  mentor
});