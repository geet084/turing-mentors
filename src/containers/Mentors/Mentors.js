import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getMentors } from '../../thunks/getMentors';
import Masonry from 'react-masonry-component';
import MentorCard from '../../components/MentorCard/MentorCard';

export class Mentors extends Component {

  componentDidMount = () => {
    const corsPrefix = 'https://cors-anywhere.herokuapp.com/';
    const root = 'https://sheltered-beach-23653.herokuapp.com';
    const path = '/api/v1/mentors?location=all';
    // TODO: what is the best way to use env vars with a rails backend?
    const url = corsPrefix + root + path
    this.props.getMentors(url);
  }

  render() {
    const { mentors } = this.props;
    const mentorList = mentors.map(mentor => {
      return <MentorCard key={mentor.id} id={mentor.id} {...mentor.attributes} />
    });

    const breakpointColumnsObj = {
      default: 4,
      1100: 3,
      700: 2,
      500: 1
    };

    return (
      <div className="mentors-container">
        <Masonry breakpointCols={breakpointColumnsObj}
          className="mentors-grid"
          columnClassName="mentors-grid_column">
          {mentorList}
        </Masonry>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  mentors: state.mentors,
})

export const mapDispatchToProps = (dispatch) => ({
  getMentors: (url) => dispatch(getMentors(url)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Mentors);