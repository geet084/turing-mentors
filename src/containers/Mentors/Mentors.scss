.mentors-container {
  padding: 10px;
}

.mentors-grid {
  display: flex;
  width: auto;
}

.mentors-grid_column {
  padding-left: 30px;
  background-clip: padding-box;
}
 
.mentors-grid_column > div {
  background: grey;
  margin-bottom: 30px;
}