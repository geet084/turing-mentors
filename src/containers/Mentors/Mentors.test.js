import React from 'react';
import { shallow } from 'enzyme';
import { Mentors, mapStateToProps, mapDispatchToProps } from './Mentors';
import { getMentors } from '../../actions';

describe('Mentors', () => {
  let wrapper;
  const mockProps = {
    getMentors: jest.fn(),
    mentors: [{ id: 1, first_name: 'first' }, { id: 2, first_name: 'second' }]
  }

  beforeEach(() => {
    wrapper = shallow(<Mentors {...mockProps} />);
  });

  describe('initial', () => {
    it('should have match the default snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have match the default snapshot with no mentors', () => {
      wrapper.setProps({ mentors: [] })
      expect(wrapper).toMatchSnapshot();
    });

    it('should have match the default snapshot with mentors', () => {
      wrapper.setProps({ mentors: [{ id: 1 }, { id: 2 }] })
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('getUpdated', () => {
    it('should retrieve the correct list of mentors in denver', () => {
      const mockURL = 'https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=denver';
      const mockEvent = { target: { name: 'denver' } };

      wrapper.instance().handleChange(mockEvent);

      expect(mockProps.getMentors).toHaveBeenCalledWith(mockURL)
    });

    it('should retrieve the correct list of mentors not in denver', () => {
      const mockURL = 'https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=remote';
      const mockEvent = { target: { name: 'remote' } };

      wrapper.instance().handleChange(mockEvent);

      expect(mockProps.getMentors).toHaveBeenCalledWith(mockURL)
    });

    it('should retrieve the correct list of mentors with javascript', () => {
      const mockURL = 'https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all&tech_skills=javascript';
      const mockEvent = { target: { name: 'javascript' } };

      wrapper.instance().handleChange(mockEvent);

      expect(mockProps.getMentors).toHaveBeenCalledWith(mockURL)
    });

    it('should retrieve the correct list of mentors with ruby', () => {
      const mockURL = 'https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all&tech_skills=ruby';
      const mockEvent = { target: { name: 'ruby' } };

      wrapper.instance().handleChange(mockEvent);

      expect(mockProps.getMentors).toHaveBeenCalledWith(mockURL)
    });
  });

  describe('handleChange', () => {
    it('should update the selected input in state', () => {
      const mockEvent = { target: { name: 'javascript' } };
      const expected = true;
      expect(wrapper.state('javascript')).toEqual(false)
      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('javascript')).toEqual(expected);
    });
  });

  describe('handleSearch', () => {
    it('should update the selected input in state', () => {
      const mockEvent = { target: { name: 'mentorSearch', value: 'search param' } };
      const expected = 'search param';
      expect(wrapper.state('mentorSearch')).toEqual('')
      wrapper.instance().handleSearch(mockEvent);

      expect(wrapper.state('mentorSearch')).toEqual(expected);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an array of mentors', () => {
      const mockState = {
        mentors: [],
        stuff: {},
        things: '',
      };

      const expected = {
        mentors: []
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call getMentors dispatch', () => {
      const mockDispatch = jest.fn();
      const mockURL = 'http://localhost:3010';
      //TODO: adjust to pass a more robust test
      // const actionToDispatch = getMentors(mockURL);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getMentors(mockURL);

      // expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});