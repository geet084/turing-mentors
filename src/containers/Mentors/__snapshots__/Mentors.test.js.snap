// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mentors initial should have match the default snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mentors
    getMentors={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    mentors={
      Array [
        Object {
          "first_name": "first",
          "id": 1,
        },
        Object {
          "first_name": "second",
          "id": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MentorControls
          c={false}
          denver={false}
          elixer={false}
          handleChange={[Function]}
          handleSearch={[Function]}
          java={false}
          javascript={false}
          mentorSearch=""
          php={false}
          python={false}
          remote={false}
          ruby={false}
          sql={false}
          swift={false}
        />,
        <MasonryComponent
          breakpointCols={
            Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            }
          }
          className="mentors-grid"
          columnClassName="mentors-grid_column"
          disableImagesLoaded={false}
          elementType="div"
          enableResizableChildren={false}
          imagesLoadedOptions={Object {}}
          onLayoutComplete={[Function]}
          onRemoveComplete={[Function]}
          options={Object {}}
          updateOnEachImageLoad={false}
        >
          <MentorCard
            id={1}
          />
          <MentorCard
            id={2}
          />
        </MasonryComponent>,
      ],
      "className": "mentors-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "c": false,
          "denver": false,
          "elixer": false,
          "handleChange": [Function],
          "handleSearch": [Function],
          "java": false,
          "javascript": false,
          "mentorSearch": "",
          "php": false,
          "python": false,
          "remote": false,
          "ruby": false,
          "sql": false,
          "swift": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "breakpointCols": Object {
            "1100": 3,
            "500": 1,
            "700": 2,
            "default": 4,
          },
          "children": Array [
            Array [
              <MentorCard
                id={1}
              />,
              <MentorCard
                id={2}
              />,
            ],
            false,
            false,
          ],
          "className": "mentors-grid",
          "columnClassName": "mentors-grid_column",
          "disableImagesLoaded": false,
          "elementType": "div",
          "enableResizableChildren": false,
          "imagesLoadedOptions": Object {},
          "onLayoutComplete": [Function],
          "onRemoveComplete": [Function],
          "options": Object {},
          "updateOnEachImageLoad": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "id": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "id": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MentorControls
            c={false}
            denver={false}
            elixer={false}
            handleChange={[Function]}
            handleSearch={[Function]}
            java={false}
            javascript={false}
            mentorSearch=""
            php={false}
            python={false}
            remote={false}
            ruby={false}
            sql={false}
            swift={false}
          />,
          <MasonryComponent
            breakpointCols={
              Object {
                "1100": 3,
                "500": 1,
                "700": 2,
                "default": 4,
              }
            }
            className="mentors-grid"
            columnClassName="mentors-grid_column"
            disableImagesLoaded={false}
            elementType="div"
            enableResizableChildren={false}
            imagesLoadedOptions={Object {}}
            onLayoutComplete={[Function]}
            onRemoveComplete={[Function]}
            options={Object {}}
            updateOnEachImageLoad={false}
          >
            <MentorCard
              id={1}
            />
            <MentorCard
              id={2}
            />
          </MasonryComponent>,
        ],
        "className": "mentors-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "c": false,
            "denver": false,
            "elixer": false,
            "handleChange": [Function],
            "handleSearch": [Function],
            "java": false,
            "javascript": false,
            "mentorSearch": "",
            "php": false,
            "python": false,
            "remote": false,
            "ruby": false,
            "sql": false,
            "swift": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "breakpointCols": Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            },
            "children": Array [
              Array [
                <MentorCard
                  id={1}
                />,
                <MentorCard
                  id={2}
                />,
              ],
              false,
              false,
            ],
            "className": "mentors-grid",
            "columnClassName": "mentors-grid_column",
            "disableImagesLoaded": false,
            "elementType": "div",
            "enableResizableChildren": false,
            "imagesLoadedOptions": Object {},
            "onLayoutComplete": [Function],
            "onRemoveComplete": [Function],
            "options": Object {},
            "updateOnEachImageLoad": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "id": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "id": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Mentors initial should have match the default snapshot with mentors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mentors
    getMentors={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all",
          ],
          Array [
            "https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all",
          ],
          Array [
            "https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    mentors={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MentorControls
          c={false}
          denver={false}
          elixer={false}
          handleChange={[Function]}
          handleSearch={[Function]}
          java={false}
          javascript={false}
          mentorSearch=""
          php={false}
          python={false}
          remote={false}
          ruby={false}
          sql={false}
          swift={false}
        />,
        <MasonryComponent
          breakpointCols={
            Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            }
          }
          className="mentors-grid"
          columnClassName="mentors-grid_column"
          disableImagesLoaded={false}
          elementType="div"
          enableResizableChildren={false}
          imagesLoadedOptions={Object {}}
          onLayoutComplete={[Function]}
          onRemoveComplete={[Function]}
          options={Object {}}
          updateOnEachImageLoad={false}
        >
          <MentorCard
            id={1}
          />
          <MentorCard
            id={2}
          />
        </MasonryComponent>,
      ],
      "className": "mentors-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "c": false,
          "denver": false,
          "elixer": false,
          "handleChange": [Function],
          "handleSearch": [Function],
          "java": false,
          "javascript": false,
          "mentorSearch": "",
          "php": false,
          "python": false,
          "remote": false,
          "ruby": false,
          "sql": false,
          "swift": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "breakpointCols": Object {
            "1100": 3,
            "500": 1,
            "700": 2,
            "default": 4,
          },
          "children": Array [
            Array [
              <MentorCard
                id={1}
              />,
              <MentorCard
                id={2}
              />,
            ],
            false,
            false,
          ],
          "className": "mentors-grid",
          "columnClassName": "mentors-grid_column",
          "disableImagesLoaded": false,
          "elementType": "div",
          "enableResizableChildren": false,
          "imagesLoadedOptions": Object {},
          "onLayoutComplete": [Function],
          "onRemoveComplete": [Function],
          "options": Object {},
          "updateOnEachImageLoad": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "id": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "id": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          false,
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MentorControls
            c={false}
            denver={false}
            elixer={false}
            handleChange={[Function]}
            handleSearch={[Function]}
            java={false}
            javascript={false}
            mentorSearch=""
            php={false}
            python={false}
            remote={false}
            ruby={false}
            sql={false}
            swift={false}
          />,
          <MasonryComponent
            breakpointCols={
              Object {
                "1100": 3,
                "500": 1,
                "700": 2,
                "default": 4,
              }
            }
            className="mentors-grid"
            columnClassName="mentors-grid_column"
            disableImagesLoaded={false}
            elementType="div"
            enableResizableChildren={false}
            imagesLoadedOptions={Object {}}
            onLayoutComplete={[Function]}
            onRemoveComplete={[Function]}
            options={Object {}}
            updateOnEachImageLoad={false}
          >
            <MentorCard
              id={1}
            />
            <MentorCard
              id={2}
            />
          </MasonryComponent>,
        ],
        "className": "mentors-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "c": false,
            "denver": false,
            "elixer": false,
            "handleChange": [Function],
            "handleSearch": [Function],
            "java": false,
            "javascript": false,
            "mentorSearch": "",
            "php": false,
            "python": false,
            "remote": false,
            "ruby": false,
            "sql": false,
            "swift": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "breakpointCols": Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            },
            "children": Array [
              Array [
                <MentorCard
                  id={1}
                />,
                <MentorCard
                  id={2}
                />,
              ],
              false,
              false,
            ],
            "className": "mentors-grid",
            "columnClassName": "mentors-grid_column",
            "disableImagesLoaded": false,
            "elementType": "div",
            "enableResizableChildren": false,
            "imagesLoadedOptions": Object {},
            "onLayoutComplete": [Function],
            "onRemoveComplete": [Function],
            "options": Object {},
            "updateOnEachImageLoad": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "id": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "id": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            false,
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Mentors initial should have match the default snapshot with no mentors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Mentors
    getMentors={
      [MockFunction] {
        "calls": Array [
          Array [
            "https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all",
          ],
          Array [
            "https://cors-anywhere.herokuapp.com/https://turing-mentors-be.herokuapp.com/api/v1/mentors?location=all",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    mentors={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <MentorControls
          c={false}
          denver={false}
          elixer={false}
          handleChange={[Function]}
          handleSearch={[Function]}
          java={false}
          javascript={false}
          mentorSearch=""
          php={false}
          python={false}
          remote={false}
          ruby={false}
          sql={false}
          swift={false}
        />,
        <MasonryComponent
          breakpointCols={
            Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            }
          }
          className="mentors-grid"
          columnClassName="mentors-grid_column"
          disableImagesLoaded={false}
          elementType="div"
          enableResizableChildren={false}
          imagesLoadedOptions={Object {}}
          onLayoutComplete={[Function]}
          onRemoveComplete={[Function]}
          options={Object {}}
          updateOnEachImageLoad={false}
        >
          <h3
            className="mentors-results"
          >
            Loading Mentors...
          </h3>
        </MasonryComponent>,
      ],
      "className": "mentors-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "c": false,
          "denver": false,
          "elixer": false,
          "handleChange": [Function],
          "handleSearch": [Function],
          "java": false,
          "javascript": false,
          "mentorSearch": "",
          "php": false,
          "python": false,
          "remote": false,
          "ruby": false,
          "sql": false,
          "swift": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "breakpointCols": Object {
            "1100": 3,
            "500": 1,
            "700": 2,
            "default": 4,
          },
          "children": Array [
            Array [],
            <h3
              className="mentors-results"
            >
              Loading Mentors...
            </h3>,
            false,
          ],
          "className": "mentors-grid",
          "columnClassName": "mentors-grid_column",
          "disableImagesLoaded": false,
          "elementType": "div",
          "enableResizableChildren": false,
          "imagesLoadedOptions": Object {},
          "onLayoutComplete": [Function],
          "onRemoveComplete": [Function],
          "options": Object {},
          "updateOnEachImageLoad": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading Mentors...",
              "className": "mentors-results",
            },
            "ref": null,
            "rendered": "Loading Mentors...",
            "type": "h3",
          },
          false,
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <MentorControls
            c={false}
            denver={false}
            elixer={false}
            handleChange={[Function]}
            handleSearch={[Function]}
            java={false}
            javascript={false}
            mentorSearch=""
            php={false}
            python={false}
            remote={false}
            ruby={false}
            sql={false}
            swift={false}
          />,
          <MasonryComponent
            breakpointCols={
              Object {
                "1100": 3,
                "500": 1,
                "700": 2,
                "default": 4,
              }
            }
            className="mentors-grid"
            columnClassName="mentors-grid_column"
            disableImagesLoaded={false}
            elementType="div"
            enableResizableChildren={false}
            imagesLoadedOptions={Object {}}
            onLayoutComplete={[Function]}
            onRemoveComplete={[Function]}
            options={Object {}}
            updateOnEachImageLoad={false}
          >
            <h3
              className="mentors-results"
            >
              Loading Mentors...
            </h3>
          </MasonryComponent>,
        ],
        "className": "mentors-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "c": false,
            "denver": false,
            "elixer": false,
            "handleChange": [Function],
            "handleSearch": [Function],
            "java": false,
            "javascript": false,
            "mentorSearch": "",
            "php": false,
            "python": false,
            "remote": false,
            "ruby": false,
            "sql": false,
            "swift": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "breakpointCols": Object {
              "1100": 3,
              "500": 1,
              "700": 2,
              "default": 4,
            },
            "children": Array [
              Array [],
              <h3
                className="mentors-results"
              >
                Loading Mentors...
              </h3>,
              false,
            ],
            "className": "mentors-grid",
            "columnClassName": "mentors-grid_column",
            "disableImagesLoaded": false,
            "elementType": "div",
            "enableResizableChildren": false,
            "imagesLoadedOptions": Object {},
            "onLayoutComplete": [Function],
            "onRemoveComplete": [Function],
            "options": Object {},
            "updateOnEachImageLoad": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading Mentors...",
                "className": "mentors-results",
              },
              "ref": null,
              "rendered": "Loading Mentors...",
              "type": "h3",
            },
            false,
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
